downstairs/src/region.rs

```diff
@@ -1229,12 +1569,137 @@ impl Region {
     }
 }
 
+/**
+ * Copy the contents of the replacement directory on to the extent
+ * files in the extent directory.
+ */
+pub fn move_replacement_extent<P: AsRef<Path>>(
+    region_dir: P,
+    eid: usize,
+) -> Result<(), CrucibleError> {
+    let destination_dir = extent_dir(&region_dir, eid as u32);
+    let extent_file_name = extent_file_name(eid as u32, None);
+    let replace_dir = replace_dir(&region_dir, eid as u32);
+    let completed_dir = completed_dir(&region_dir, eid as u32);
+
+    // XXX replace panic with CrucibleError
+    assert!(Path::new(&replace_dir).exists());
+    assert!(!Path::new(&completed_dir).exists());
+
+    println!("Move stuff from {:?} in {:?}", replace_dir, destination_dir,);
+
+    // Setup the original and replacement file names.
+    let mut new_file = replace_dir.clone();
+    new_file.push(extent_file_name.clone());
+
+    let mut original_file = destination_dir;
+    original_file.push(extent_file_name);
+
+    // Copy the new file (the one we copied from the source side) on top
+    // of the original file.
```

Comment:
do we do this rather than linking because we may have the file open?

Reply:
What you said makes sense. I mean like the `link(2)` system call to create a hard link.